### PR TITLE
add '-D' flag to support reading request body from file

### PR DESCRIPTION
**Pull request by zjlletian**
add '-D' flag to support reading request body from file.
for example, -D /root/reqbody, -D ../reqbody

### CODE REVIEW

File: hey.go [hunk 1]
```diff
@@ -189,6 +192,15 @@ func main() {
 		req.Host = *hostHeader
 	}
 
+	// if request body is file
+	if *bodyPath != "" {
+		if bytes, err := ioutil.ReadFile(*bodyPath); err != nil {
+			usageAndExit(err.Error())
```
Comment (rakyll) [inline hey.go]: dont print the usage here. log.Fatal(err) maybe.

File: hey.go [hunk 2]
```diff
@@ -189,6 +192,15 @@ func main() {
...
+		} else {
```
Comment (rakyll) [inline hey.go]: you dont need an else.

*body = string(all)

File: hey.go [hunk 3]
```diff
@@ -189,6 +192,15 @@ func main() {
 		req.Host = *hostHeader
 	}
 
+	// if request body is file
+	if *bodyPath != "" {
+		if bytes, err := ioutil.ReadFile(*bodyPath); err != nil {
```
Comment (rakyll) [inline hey.go]: call it all

File: hey.go (hunk 2 above)
Comment (zjlletian) [inline hey.go]: modified

File: hey.go [hunk 4]
```diff
@@ -89,6 +92,7 @@ Options:
   -t  Timeout in seconds. Default is 20, use 0 to disable.
   -A  HTTP Accept header.
   -d  HTTP request body.
+  -D  HTTP request body from file, for example, -D /root/reqbody, -D ../reqbody
```
Comment (rakyll) [inline hey.go]: Can we make the sample path more obvious?

-D HTTP request body from file, for example, /home/user/file.txt or ./file.txt

File: hey.go [hunk 5]
```diff
@@ -189,6 +193,15 @@ func main() {
 		req.Host = *hostHeader
 	}
 
+	// if request body is file
+	if *bodyPath != "" {
+		all, err := ioutil.ReadFile(*bodyPath)
+		if err != nil {
+			 log.Fatal(err)
+		}
+		*body = string(all)
```
Comment (rakyll) [inline hey.go]: I wonder if we should modify RequestBody to be a byte slice rather than casting all to string.

**Comment by rakyll** [comment]
Closed via 07ebe6726d667e01b795276b1f874844947fc0ba.

**Comment by polefishu** [comment]
how to specified a variety of request body, try [hey -n 20 -c 2 -t 50 -m post -T "application/json" -D 1.txt -D 2.txt http://localhost:2333/api/xxx], but it don't work.

**Pull request closed without merging**